12Lesson/classes.py

```diff
@@ -0,0 +1,72 @@
+# 1 задание
+from dataclasses import dataclass
+
+
+class BookValueError(Exception):
+    pass
+
+
+@dataclass(order=True)
+class Book:
+    pages: int
+    year: int
+    author: str
+    price: float
+    book_id: int = None
+
+    def __post_init__(self):
+        if self.pages <= 0 or not isinstance(self.pages, int):
```

Comment:
class InvalidPageCountError(Exception):
    pass


class InvalidYearError(Exception):
    pass


class InvalidPriceError(Exception):
    pass

Reply:
we can create more errors and use a separate error for each case